### PR TITLE
Keras updates

### PR DESCRIPTION
**Pull request by thomas-a-neil**
Confirmed that train_vae.ipynb can run with the following changes

### CODE REVIEW

File: zero_shot_sampling/msa_vae.py [hunk 1]
```diff
@@ -112,7 +112,11 @@ def load(
       **architecture_hparams,
   )
 
-  model.load_weights(path.join(model_path, 'weights'))
+  # Build the model with a dummy input before loading weights
+  dummy_input = np.zeros((1, len(homolog_config.match_state_parent)), dtype=np.int32)
```
Comment (davidBelanger) [inline zero_shot_sampling/msa_vae.py:116]: can you call this 'placeholder_input'?

File: zero_shot_sampling/msa_vae.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def save(
   """Saves a VAE model and its metadata to disk."""
   utils.make_directory(model_path)
 
-  model.save_weights(path.join(model_path, 'weights'))
+  model.save_weights(path.join(model_path, 'model.weights.h5'))
```
Comment (davidBelanger) [inline zero_shot_sampling/msa_vae.py:89]: iirc the advantage of saving in the default tf format instead of h5 is that the model is self-describing; we don't need to build it before restoring the weights. Did things not work with the default file format?

Comment (davidBelanger) [reply zero_shot_sampling/msa_vae.py:89]: Based on [1], this is due to using keras v3 vs. v2. It's nice to have things work with v3, so let's stick to your approach rather than pinning things to an older version of keras.

[1] https://discuss.ai.google.dev/t/how-to-properly-save-and-load-model-weights/23916
